### ISSUE TITLE
navigateTo causes hydration errors from route middleware (client only)

### ISSUE THREAD

**Issue by Kapcash** [nuxt/nuxt.js#15136]

**Comment by Ed1ks** [comment]
Any updates on this? I am having the same problem that I want to redirect to auth page on initial load, but cant. And there doesnt seem to be a fix avaliable.

**Comment by danielroe** [comment]
You can redirect on the server or after the app is hydrated, such as in onMounted or onNuxtReady.

**Comment by gerbenvandijk** [comment]
@danielroe sorry to bring up and old issue, but I was wondering if this is something that can be reconsidered. 

it's kind of painful to have to redirect in onMounted e.g. for authentication guards. most of our app is SSR, except for some auth related pages. having to redirect in the onMounted makes these pages needlessly complex. e.g. we have to hide the UI until we know if the user had to be redirected or not. then we need to wrap all other code in onMounted in that same check. if we want to do that in a clean way on multiple pages, we also have to make composables for those things (should it redirect, the actual redirect). etc. compared to a single line of code pointing to a middleware that redirects before all the other page logic even runs, it is a lot of hassle. also in Nuxt2 this was working on both the server and clientside perfectly fine without any hydration issues.

**Comment by gerbenvandijk** [comment]
@danielroe Thanks a bunch for your time!! 

We had tried this before and still gotten hydration errors - however due to your comment we were able to find the cause.

It was a stupid thing, my routeRules were not applied as expected due to our use of the `@nuxtjs/i18n` module. After changing my routeRules to e.g. 

```
'*/account/**': {
    ssr: false
}
```
in stead of 
```
'account/**': {
    ssr: false
}
```
everything was working as expected! Thanks a bunch 👍 👍 👍
